### PR TITLE
feat(flow): Adds flow-item component to replace use of panel within a flow

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #5071

## Summary

feat(flow): Adds flow-item component to replace use of panel within a flow. (#5071)

### CODE REVIEW

File: src/components/panel/panel.scss [hunk 1]
```diff
@@ -178,3 +166,7 @@
   z-sticky;
   inline-size: fit-content;
 }
+
+[hidden] {
```
Comment (driskull) [inline src/components/panel/panel.scss:170]: Maybe this could be something we include in every component somehow?

File: src/components/flow/flow.tsx [hunk 2]
```diff
@@ -81,50 +81,52 @@ export class Flow {
...
-    const activePanel = newPanels[newPanelCount - 1];
-    const previousPanel = newPanels[newPanelCount - 2];
-
-    if (newPanelCount && activePanel) {
-      newPanels.forEach((panelNode) => {
-        panelNode.showBackButton = panelNode === activePanel && newPanelCount > 1;
-        panelNode.hidden = panelNode !== activePanel;
+    const { el, flowItems } = this;
+
+    const newFlowItems: HTMLCalciteFlowItemElement[] = Array.from(
```
[the remark this answers is not in the document]
Comment (driskull) [reply src/components/flow/flow.tsx:103]: That would be great. thanks!

File: src/components/flow/flow.tsx [hunk 3]
```diff
@@ -50,13 +50,13 @@ export class Flow {
 
   @Element() el: HTMLCalciteFlowElement;
 
-  @State() panelCount = 0;
+  @State() flowItemCount = 0;
 
   @State() flowDirection: FlowDirection = null;
 
-  @State() panels: HTMLCalcitePanelElement[] = [];
+  @State() flowItems: HTMLCalciteFlowItemElement[] = [];
```
[the remark this answers is not in the document]
Comment (driskull) [reply src/components/flow/flow.tsx:57]: Makes sense

File: src/components/panel/panel.tsx [hunk 4]
```diff
@@ -68,6 +64,8 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
 
   /**
    * When provided, this method will be called before it is removed from the parent flow.
+   *
+   *  @deprecated use `calcite-flow-item` instead.
```
[the remark this answers is not in the document]
Comment (driskull) [reply src/components/panel/panel.tsx:68]: Agreed. We can support both for a while I guess.

File: src/components/panel/panel.scss (hunk 1 above)
Comment (driskull) [reply src/components/panel/panel.scss:170]: Seems that PR was already merged. I think this was still needed. Might need some investigation. 🔍

File: src/components/flow/flow.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/flow/flow.tsx]: Pushed changes, can you take a look? 👀

File: src/components/flow-item/flow-item.tsx [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+          headingLevel={headingLevel}
+          heightScale={heightScale}
+          intlBack={intlBack}
+          intlClose={intlClose}
+          intlOptions={intlOptions}
+          loading={loading}
+          menuOpen={menuOpen}
+          widthScale={widthScale}
+        >
+          <slot name={SLOTS.headerActionsStart} slot={PANEL_SLOTS.headerActionsStart} />
```
Comment (driskull) [inline src/components/flow-item/flow-item.tsx:258]: @jcfranco can you revert the changes you made using `getSlotted()`? That won't work with this because getSlotted will always return true in this case.

We should not use getSlotted anymore because it won't work if nested slots exist.

**Comment by macandcheese** [comment]
@geospatialem @benelan - we'll need to add a new page to docs site for this, with some guidelines / samples if possible.

File: src/components/flow-item/flow-item.tsx (hunk 5 above)
Comment (jcfranco) [reply src/components/flow-item/flow-item.tsx:258]: Sorry, I unintentionally rolled back changes to panel. Can you take another look?

Comment (driskull) [reply src/components/flow-item/flow-item.tsx:258]: Changes look good now! ✨

File: src/components/flow/flow.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (driskull) [reply src/components/flow/flow.tsx]: Changes look good now! ✨

File: src/components/flow/flow.e2e.ts [hunk 6]
```diff
@@ -21,214 +22,245 @@ describe("calcite-flow", () => {
...
-    await page.$eval(
-      "calcite-panel",
-      (elm: HTMLCalcitePanelElement) =>
-        (elm.beforeBack = (window as typeof window & Pick<typeof elm, "beforeBack">).beforeBack)
-    );
+      it("goes back when item back button is clicked", async () => {
+        const page = await newE2EPage();
 
-    const flow = await page.find("calcite-flow");
+        await page.setContent(`<calcite-flow><${itemTag}></${itemTag}></calcite-flow>`);
```
Comment (driskull) [inline src/components/flow/flow.e2e.ts:52]: I think this needs 2 itemTag types for back button to show up.

Comment (jcfranco) [reply src/components/flow/flow.e2e.ts]: Good catch. Fixed.

**Comment by jcfranco** [comment]
Non-breaking tweaks have been pushed. 

@driskull Can you take a look at the screenshot differences? 🙇

**Review by jcfranco — APPROVED** [review]
〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️
〰️🌊🌊🌊🌊〰️🌊〰️〰️〰️〰️🌊🌊〰️〰️🌊〰️〰️〰️〰️〰️🌊〰️
〰️🌊〰️〰️〰️〰️🌊〰️〰️〰️🌊〰️〰️🌊〰️🌊〰️〰️〰️〰️〰️🌊〰️
〰️🌊🌊🌊〰️〰️🌊〰️〰️〰️🌊〰️〰️🌊〰️🌊〰️〰️🌊〰️〰️🌊〰️
〰️🌊〰️〰️〰️〰️🌊〰️〰️〰️🌊〰️〰️🌊〰️🌊〰️🌊〰️🌊〰️🌊〰️
〰️🌊〰️〰️〰️〰️🌊🌊🌊〰️〰️🌊🌊〰️〰️〰️🌊〰️〰️〰️🌊〰️〰️
〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️〰️

**Comment by driskull** [comment]
@jcfranco screenshots look ok to me. Do we need anyone else to review them?

**Comment by driskull** [comment]
@jcfranco good to merge?

File: src/components/flow/flow.e2e.ts [hunk 7]
```diff
@@ -21,214 +22,249 @@ describe("calcite-flow", () => {
...
-    await page.setContent("<calcite-flow><calcite-panel></calcite-panel></calcite-flow>");
+        expect(flowItem).toBeNull();
+      });
 
-    await page.$eval(
-      "calcite-panel",
-      (elm: HTMLCalcitePanelElement) =>
-        (elm.beforeBack = (window as typeof window & Pick<typeof elm, "beforeBack">).beforeBack)
-    );
+      it.only("goes back when item back button is clicked", async () => {
```
Comment (driskull) [inline src/components/flow/flow.e2e.ts:49]: @jcfranco is the `it.only` a whoopsie? 😱

Comment (jcfranco) [reply src/components/flow/flow.e2e.ts:49]: 100% whoopsie right there. Removing! ✨🧹✨

**Comment by driskull** [comment]
@jcfranco I think this one good to go

**Comment by jcfranco** [comment]
273a7ba520f3cc9239b13bc6cee72841
